lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +136,70 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+  TODO: how would we represent multiplicity of roots
+*/
+
+function getSolutionsAndSymbol (equation) {
+  const left = equation.leftNode;
+  const rightNode = Node.Creator.constant(0);
+
+  const solutions = [];
+
+  let symbol, steps, factorsWithSymbols;
+
+  // If left hand side is a power node and it does not resolve to a constant (e.g. 2^7)
+  // then it is a factor. If left hand side is a multiplication node, return a list of all the valid factors.
+  if (Node.Type.isOperator(left, '^') && !checks.resolvesToConstant(left)){
+    factorsWithSymbols = [left];
+  }
+  else {
+    factorsWithSymbols = equation.leftNode.args.filter(arg => !checks.resolvesToConstant(arg));
+  }
+
+  // For each factor, solve the equation factor = 0 for the symbol
+  // If the factor is a power node, solve the equation base = 0 for the symbol
+  for (var f in factorsWithSymbols) {
+    let factor = factorsWithSymbols[f];
+
+    if (Node.Type.isOperator(factor, '^')) {
+      factor = factor.args[0];
+    }
+
+    const leftNode = Node.Type.isParenthesis(factor)
+      ? factor.content
+      : factor;
+
+    steps = stepThrough(leftNode, rightNode, '=');
+
+    if (steps.length > 0) {
+      // e.g. x - 2 = 0
+      const lastStep = steps.slice(-1)[0];
+
+      symbol = lastStep.newEquation.leftNode;
+
+      // Append to a list of roots
+      if (Node.Type.isSymbol(symbol)) {
+        solutions.push(lastStep.newEquation.rightNode);
+      }
+    }
+    else {
+      // e.g. x = 0
```

Comment:
this is really small and not required, but based on Lili's comments and the way I read it too, it might be easier to read if you do 

`if (steps.length === 0)` and put this code first, and then do the !== 0 case as the else afterwards, so that the if/else are closer (totally optional)